cupy/core/core.pyx

```diff
@@ -55,12 +56,10 @@ from cupy_backends.cuda.libs cimport cublas
 
 @cython.profile(False)
 cdef inline _should_use_rop(x, y):
-    xp = getattr(x, '__array_priority__', 0)
-    yp = getattr(y, '__array_priority__', 0)
-    return xp < yp and not isinstance(y, ndarray)
+    return not isinstance(y, _KNOWN_TYPES) and hasattr(y, '__array_ufunc__')
```

Comment:
> Would you check the CI failure? It's about binary operations of sparse matrices.

`cupy.ndarray op cupyx.sparse.spmatrix` should use rop.  `spmatrix` does not have `__array_ufunc__`.